migrations/migration.go

```diff
@@ -0,0 +1,159 @@
+package migrations
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// A Migration performs a schema migration on the database.
+type Migration interface {
+	// DoMigration runs the actual migration. All database
+	// access must be through the transaction given as an
+	// argument. It should not manually commit or roll
+	// back the transaction.
+	DoMigration(*sqlx.Tx) error
+}
+
+type NamedMigration struct {
+	Name      string
+	Migration Migration
+}
+
+// A Migration that is just a list of SQL statements to perform. The
+// DoMigration method will execute all the SQL.
+type StaticMigration []string
```

Comment:
As we discussed in Slack, let's add the `customMigration` type back in, more to communicate that possibility to users of the library since it's a trivial implementation